pkg/northbound/gnmi/set.go

```diff
@@ -428,3 +434,18 @@ func validateChange(target string, deviceType devicetype.Type, version devicetyp
 	log.Infof("Validating change %s:%s:%s DONE", target, deviceType, version)
 	return nil
 }
+
+func getProtocolState(device *topodevice.Device) topodevice.ServiceState {
```

Comment:
I saw something like this in Jordan's code the other day - hope it will be reused

Reply:
that was the channel, this is the service. at the moment I'm figuring let's just put it in then we can refactor.